components/Tabs/examples/MDCTabBarViewTypicalExampleViewController.m

```diff
@@ -265,6 +271,65 @@ - (void)changeItemsToTextAndImage {
   }
 }
 
+#pragma mark - UIScrollViewDelegate
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+  [self logItemVisibilityChanges];
+}
+
+#pragma mark - UIViewController
+
+- (void)viewWillTransitionToSize:(CGSize)size
+       withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  [coordinator
+      animateAlongsideTransition:nil
+                      completion:^(
+                          id<UIViewControllerTransitionCoordinatorContext> _Nonnull context) {
+                        [self logItemVisibilityChanges];
+                      }];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self logItemVisibilityChanges];
+}
+
+- (void)logItemVisibilityChanges {
+  NSMutableSet<UITabBarItem *> *visibleItems = [NSMutableSet set];
+  for (UITabBarItem *item in self.tabBar.items) {
+    CGRect itemViewInWindow = [self.tabBar rectForItem:item inCoordinateSpace:self.view.window];
+    CGRect overlapRect = CGRectIntersection(self.view.window.bounds, itemViewInWindow);
+
+    // Avoid dividing by zero
+    if (CGRectIsNull(overlapRect) || MDCCGFloatEqual(CGRectGetWidth(itemViewInWindow), 0)) {
+      continue;
+    }
+    [visibleItems addObject:item];
+  }
+
+  NSMutableSet<UITabBarItem *> *noLongerVisibleItems = [NSMutableSet set];
+  NSMutableSet<UITabBarItem *> *visibleItemsCopy = [visibleItems copy];
+  for (UITabBarItem *item in self.visibleItems) {
+    if (![visibleItems containsObject:item]) {
+      [noLongerVisibleItems addObject:item];
+    }
+    [visibleItems removeObject:item];
+  }
+  self.visibleItems = visibleItemsCopy;
```

Comment:
I got a little lost on these loops.

Why do you assign the visibleItemsCopy back to self.visibleItems on line 319? Wouldn't you want to assign `visibleItems` since it now contains only the still-visible items.

It seems `visibleItemsCopy` would still contain the elements in `noLongerVisibleItems`.

Reply:
I rewrote the logic and renamed the sets. Let me know what you think.